pkg/kdump/test-config-client.js

```diff
@@ -44,37 +44,53 @@ const changedConfig = [
     ""
 ].join("\n");
 
-QUnit.test("config_update", function (assert) {
-    const done = assert.async();
-    assert.expect(6);
-    const dataWasChanged = new Promise(resolve => { this.dataWasChangedResolve = resolve });
-    let config;
-    const configChanged = (event, settings) => {
-        assert.equal(settings._internal.foo.value, "moo", "value changed correctly");
-        assert.equal("key" in settings._internal, false, "setting with comment deleted correctly");
-        assert.equal("will" in settings._internal, false, "setting without comment deleted correctly");
-        assert.equal(settings._internal.hooray.value, "value", "value added correctly");
-        assert.equal(config._rawContent, changedConfig, "raw text for changed config is correct");
-        this.dataWasChangedResolve();
-    };
+QUnit.module("kdump", hooks => {
+    let filename = "";
 
-    const filename = "cockpit_config_read";
-    const configFile = cockpit.file(filename);
-    configFile
-            .replace(basicConfig)
-            .then(() => {
-                assert.equal(configFile.path, filename, "file has correct path");
-                config = new kdump.ConfigFile(filename);
-                config.wait().then(() => {
-                    config.settings._internal.foo.value = "moo";
-                    delete config.settings._internal.key;
-                    delete config.settings._internal.will;
-                    config.settings._internal.hooray = { value: "value" };
-                    config.addEventListener('kdumpConfigChanged', configChanged);
-                    config.write(config.settings)
-                            .then(() => dataWasChanged.then(done));
+    hooks.before(async () => {
+        const resp = await cockpit.spawn(["bash", "-c", "echo $(mktemp --suffix kdump-test)"]);
```

Comment:
Can you please drop the bash and echo wrappers? Just calling `mktemp` and grabbing its output should be exactly right.